### ISSUE TITLE
appendToBody property breaks date selection in SingleDatePicker

### ISSUE THREAD

**Issue by gilbsgilbs** [airbnb/react-dates#1535]
**react-dates version**

- react-dates@19.0.0
- react-dates@18.5.x
- react-dates@18.4.1

**Describe the bug**

When `appendToBody` is set in a `SingleDatePicker`, the picker displays correctly on click but `onDateChange` callback is never called and the picker gets closed immediately. The date doesn't update.

This actually looks like a regression from #1522 . It appears that it was introduced in 18.4.1 (18.4.0 and previous versions work as expected).

**Source code (including props configuration)**

```jsx
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      focused: props.autoFocus,
      date: props.initialDate,
    };

    this.onDateChange = this.onDateChange.bind(this);
    this.onFocusChange = this.onFocusChange.bind(this);
  }

  onDateChange(date) {
    this.setState({ date });
  }

  onFocusChange({ focused }) {
    this.setState({ focused });
  }

  render() {
   const { focused, date } = this.state; 
    return (
        <SingleDatePicker
          id="mydatepicker"
          date={date}
          focused={focused}
          onDateChange={this.onDateChange}
          onFocusChange={this.onFocusChange}
          appendToBody
        />
    );
  }
}
```

**Screenshots/Gifs**

None.

**Desktop (please complete the following information):**
 - OS: Archlinux
 - Browser: Chromium, Firefox
 - Version: 72, 65

**Is the issue reproducible in Storybook?**

I don't know how to use storybook.

**Additional context**

Let me know if I can do anything to help.

**Comment by danedavid** [comment]
I tried reverting https://github.com/airbnb/react-dates/pull/1533, which was a fix for https://github.com/airbnb/react-dates/issues/1522, but issue still persists.

**Comment by gilbsgilbs** [comment]
Hi,

Thanks @danedavid for the PR. Although reproduction source code from first post indeed seems fixed by #1540, the issue persists in a real-world project. Switching back to 18.4.0 works fine, while 19.0.3 breaks consistently. I'm still trying hard to figure out what makes the bug happen, but I'm currently failing to have a clean repro I can show. It may be a conflict with some other component because it's only reproducing in one specific place of the project and works normally elsewhere.

Any idea? Worth reopening IMHO.

**Comment by ljharb** [comment]
@gilbsgilbs filing a new issue with the repro, when you can find it, would be most effective.
